### PR TITLE
chore(cloudfoundry): Upgrades OkHttp and Retrofit libraries, making a…

### PR DESCRIPTION
**Pull request by rvazquezglez**
chore(cloudfoundry): Upgrades OkHttp and Retrofit libraries, making all the changes on cloudfoundry to keep it working as before.

### CODE REVIEW

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java [hunk 1]
```diff
@@ -78,7 +99,9 @@ public ForkJoinPool cloudFoundryThreadPool(
                 cacheRepository,
                 a.getPermissions().build(),
                 cloudFoundryThreadPool,
-                a.getSpaceFilter()),
+                a.getSpaceFilter(),
+                new OkHttpClient(),
```
Comment (zachsmith1) [inline clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java:103]: We should be using the new okhttpclient from the method above. Check line 80 as we are already autowiring it as a method parameter

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java [hunk 2]
```diff
@@ -78,7 +99,9 @@ public ForkJoinPool cloudFoundryThreadPool(
...
+                new SimpleMeterRegistry()),
```
Comment (zachsmith1) [inline clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java:104]: Same as last comment. Lets use the autowired meter (line 81)

**Review by zachsmith1 — COMMENTED** [review]
See comments -- still reviewing

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java (hunk 1 above)
Comment (rvazquezglez) [reply clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/config/CloudFoundryProviderConfig.java]: Changed!

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java [hunk 3]
```diff
@@ -104,157 +67,115 @@
...
-    okHttpClient.interceptors().add(this::createRetryInterceptor);
-
     ObjectMapper mapper = new ObjectMapper();
     mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
     mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
     mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
     mapper.registerModule(new JavaTimeModule());
 
-    this.jacksonConverter = new JacksonConverter(mapper);
+    okHttpClientBuilder.eventListener(
```
Comment (zachsmith1) [inline clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java:94]: Let's remove the eventListener for now (also un-autowire the registry/remove it). I found a more "spinnaker" way of doing it and we will be able to leverage OkHttp3MetricsInterceptor from kork-web. However, lets do that in a separate PR.

**Review by zachsmith1 — CHANGES_REQUESTED** [review]
See comments

File: clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java (hunk 3 above)
Comment (rvazquezglez) [reply clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java]: Done

**Review by zachsmith1 — APPROVED** [review]
LGTM

**Review by german-muzquiz — APPROVED** [review]
